### DISCUSSION TITLE
zh/help/FAQ/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#8430] [Docs]
# zh/help/FAQ/

查找Ultralytics YOLO  相关常见问题的解决方案。了解硬件要求、微调YOLO 模型、转换为ONNX/TensorFlow 等信息。

https://docs.ultralytics.com/zh/help/FAQ/

**Comment by Zhou2518** [comment]
When add SwinTransformer ,Fewer epochs operate normally, and when epochs are larger, errors are reported in the final val phase：../aten/src/ATen/native/cuda/ScatterGatherKernel.cu:144: operator(): block: [630,0,0], thread: [32,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.
../aten/src/ATen/native/cuda/ScatterGatherKernel.cu:144: operator(): block: [630,0,0], thread: [33,0,0] Assertion `idx_dim >= 0 && idx_dim < index_size && "index out of bounds"` failed.

**Reply by pderrenger** [reply]
@Zhou2518 it seems like you're encountering an index out-of-bounds error during the validation phase when using SwinTransformer with a larger number of epochs. This could be related to the dataset or a bug in the code.

Here's a quick checklist to troubleshoot the issue:
1. Ensure your dataset annotations are correct and within the image bounds.
2. Verify that the dataset loader is not returning any invalid indices.
3. Check if there's an update or bug fix for SwinTransformer that addresses this issue.

If the problem persists, consider opening an issue on the Ultralytics YOLO GitHub repository with detailed error logs and your training setup. This way, the community can help you out more effectively. Meanwhile, you might want to try reducing the batch size or image size to see if it mitigates the issue.

For example, if you're using the CLI, you could adjust the batch size like so:
```bash
yolo detect val model=yolov8n.pt data=coco128.yaml batch=8 imgsz=640
```

Hope this helps! If you need further assistance, don't hesitate to reach out. 😊👍

**Comment by why5602337** [comment]
What's the difference between yolov9's SPPELAN and v8's SPPF? It feels the same structure. Why is the number of parameters reduced

**Reply by glenn-jocher** [reply]
Hey there! 😊 The SPPELAN in YOLOv9 and SPPF in YOLOv8 might seem structurally similar because they both aim to enhance feature extraction by aggregating contextual information at various scales. However, the reduction in the number of parameters in YOLOv9's SPPELAN compared to YOLOv8's SPPF is achieved through more efficient design choices and optimizations that maintain or improve performance while using fewer resources. These optimizations could include alterations in layer configurations, filter sizes, or the integration method of multi-scale features. The goal is always to strike a balance between model size, speed, and accuracy. If you're curious about the specific technical details, diving into the source code or architecture diagrams might offer deeper insights! 🚀

**Reply by MoAbbasid** [reply]
@why5602337 unluckily you got a bot response, if you figured out the difference be sure to let us know.
